sw/device/tests/rstmgr_alert_info_test.c

```diff
@@ -0,0 +1,686 @@
+        },
+    [kTopEarlgreyAlertPeripheralKmac] =
+        {
+            .name = "KMAC",
+            .alert = kTopEarlgreyAlertIdKmacFatalFaultErr,
+            .class = kDifAlertHandlerClassB,
+        },
+    [kTopEarlgreyAlertPeripheralUart0] =
+        {
+            .name = "UART0",
+            .alert = kTopEarlgreyAlertIdUart0FatalFault,
+            .class = kDifAlertHandlerClassC,
+        },
+    [kTopEarlgreyAlertPeripheralUart1] =
+        {
+            .name = "UART1",
+            .alert = kTopEarlgreyAlertIdUart1FatalFault,
+            .class = kDifAlertHandlerClassC,
+        },
+    [kTopEarlgreyAlertPeripheralUart2] =
+        {
+            .name = "UART2",
+            .alert = kTopEarlgreyAlertIdUart2FatalFault,
+            .class = kDifAlertHandlerClassC,
+        },
+    [kTopEarlgreyAlertPeripheralUart3] =
+        {
+            .name = "UART3",
+            .alert = kTopEarlgreyAlertIdUart3FatalFault,
+            .class = kDifAlertHandlerClassC,
+        },
+    [kTopEarlgreyAlertPeripheralI2c0] =
+        {
+            .name = "I2C0",
+            .alert = kTopEarlgreyAlertIdI2c0FatalFault,
+            .class = kDifAlertHandlerClassA,
+        },
+    [kTopEarlgreyAlertPeripheralI2c1] =
+        {
+            .name = "I2C1",
+            .alert = kTopEarlgreyAlertIdI2c1FatalFault,
+            .class = kDifAlertHandlerClassA,
+        },
+    [kTopEarlgreyAlertPeripheralI2c2] = {
+        .name = "I2C2",
+        .alert = kTopEarlgreyAlertIdI2c2FatalFault,
+        .class = kDifAlertHandlerClassA,
+    }};
+
+bool test_main(void) {
+  // Enable global and external IRQ at Ibex.
+  irq_global_ctrl(true);
+  irq_external_ctrl(true);
+
+  CHECK_DIF_OK(dif_rstmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_RSTMGR_AON_BASE_ADDR), &rstmgr));
+  CHECK_DIF_OK(dif_alert_handler_init(
+      mmio_region_from_addr(TOP_EARLGREY_ALERT_HANDLER_BASE_ADDR),
+      &alert_handler));
+  CHECK_DIF_OK(dif_rv_plic_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR), &plic));
+
+  peripheral_init();
+
+  // Enable all interrupts used in this test.
+  rv_plic_testutils_irq_range_enable(&plic, kPlicTarget,
+                                     kTopEarlgreyPlicIrqIdAlertHandlerClassa,
+                                     kTopEarlgreyPlicIrqIdAlertHandlerClassd);
+  rv_plic_testutils_irq_range_enable(
+      &plic, kPlicTarget, kTopEarlgreyPlicIrqIdAonTimerAonWkupTimerExpired,
+      kTopEarlgreyPlicIrqIdAonTimerAonWdogTimerBark);
+
+  // endable alert info
+  CHECK_DIF_OK(dif_rstmgr_alert_info_set_enabled(&rstmgr, kDifToggleEnabled));
+
+  // Check if there was a HW reset caused by the escalation.
+  dif_rstmgr_reset_info_bitfield_t rst_info;
+  CHECK_DIF_OK(dif_rstmgr_reset_info_get(&rstmgr, &rst_info));
+  CHECK_DIF_OK(dif_rstmgr_reset_info_clear(&rstmgr));
+
+  LOG_INFO("reset info = 0x%02X", rst_info);
+  global_alert_called = 0;
+
+  if (rst_info == kDifRstmgrResetInfoPor) {
+    global_test_round = kRound1;
```

Comment:
if you want to get really fancy, you could technically store `global_test_round` in flash and read it back to see if it matches your expectations :)

Reply:
noted, I can update that after we implement flash write/read back test